src/Tickets/Commerce/Flag_Actions/Flag_Action_Handler.php

```diff
@@ -69,6 +74,28 @@ public function register() {
 		}
 
 		$this->container->singleton( static::class, $this );
+
+		add_filter( 'tec_tickets_commerce_prepare_order_for_email_send_email_completed_order', [ $this, 'prepare_order_for_email' ] );
+		add_filter( 'tec_tickets_commerce_prepare_order_for_email_send_email_purchase_receipt', [ $this, 'prepare_order_for_email' ] );
+	}
+
+	/**
+	 * Prepare the order for email.
+	 *
+	 * @since TBD
+	 *
+	 * @param WP_Post $order The order to prepare.
+	 *
+	 * @return WP_Post
+	 */
+	public function prepare_order_for_email( WP_Post $order ) {
+		if ( empty( $order->items ) || ! is_array( $order->items ) ) {
+			return $order;
+		}
+
+		$order->items = array_filter( $order->items, fn ( $item ) => $this->is_ticket( $item['product_id'] ) );
```

Comment:
`is_ticket()` need the entire `$item` array. This change caused a fatal error on my local that interrupted the process of placing an order.

Also, while doing a debug, it does not appear that the `product_id` property exists.

```suggestion
		$order->items = array_filter( $order->items, fn ( $item ) => $this->is_ticket( $item ) );
```

Reply:
Nice this also needs tests